pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,829 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import logging
+import re
+import sys
+
+import numpy as np
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+# noinspection DuplicatedCode
```

Comment:
Ha, is PyCharm trying to tell you something? :-) Maybe this should not be copied, but refactored so it's usable by both instruments? 
With every instrument we take on a maintenance burden of sometimes >1k lines of code, if that can be avoided, it should be!
Nowadays we can use [dynamic properties](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html#dynamic-properties) to selectively replace some elements of a property definition (e.g. the command syntax).

Reply:
This could be something to consider if the delta is not too large.
In any case, good that you reused property/method names, that will make it easier to create "common" APIs per instrument class (#416)

Reply:
Yes that PyCharm telling me that the `Channel` class of LeCrosy T3DSO1204  is similar to the `Channel` class of Keysight DSOX1102G. If you want I can delete that comment.

As for refactoring the upper layer to avoid code duplication, I could do it but it would be outside of the scope of this pull request and I would need some time and guidance.

Reply:
Nah, the comment doesn't bother me, all that copy-and-nearly-paste triggers my gut feelings. But I see how it's barely avoidable. :-/ 
Let's proceed like this for now. If you can make the/a number of tests hardware-independent, that will make that refactoring that much easier as one has a source of truth to compare against.

Reply:
I see your point. Anyway I have already made a number of tests hardware-independent.